xcell/mappers/mapper_DESY3wl.py

```diff
@@ -0,0 +1,446 @@
+
+    def get_nz(self, dz=0):
+        """
+        Returns the mappers redshift distribtuion of sources from a file.
+
+        Kwargs:
+            dz=0
+
+        Returns:
+            [z, nz] (Array)
+        """
+        if self.dndz is None:
+            bn = f"BIN{self.zbin + 1}"
+            f = fitsio.read(self.config['file_nz'], ext="nz_source",
+                            columns=["Z_MID", bn])
+            self.dndz = {'z_mid': f['Z_MID'], 'nz': f[bn]}
+        return self._get_shifted_nz(dz)
+
+    def _get_mask(self):
+        # Returns the  mask.
+
+        pos = self.get_positions()
+        msk = get_map_from_points(pos, self.nside, w=self.get_weights(),
+                                  ra_name='ra', dec_name='dec',
+                                  rot=self.rot)
+        # Removing the overlapping area at the mask level. Note that this is an
+        # approximation and weights, biases, etc should be recomputed.
+        if self.remove_overlap is not None:
+            for v in self.remove_overlap.values():
+                m = hp.read_map(v)
+                m[m == hp.UNSEEN] = 0
+                m = rotate_mask(m, self.rot)
+                m = hp.ud_grade(m, nside_out=self.nside)
+                # Set filled pixels to 0
+                msk[m != 0] = 0
+        return msk
+
+    def get_nl_coupled(self, mode=None):
+        e1f, e2f, mod = self._set_mode(mode)
+        if self.nls[mod] is not None:
+            self.nl_coupled = self.nls[mod]
+            return self.nl_coupled
+
+        # This will only be computed if self.nls['mod'] is None
+        def get_w2s2():
+            pos = self.get_positions()
+            weights = self.get_weights()
+            ellips = self.get_ellips_unbiased(mod)
+            w = 0.5*np.sum(ellips**2, axis=0) * weights**2
+            mp = get_map_from_points(pos, self.nside, w=w,
+                                     ra_name='ra', dec_name='dec',
+                                     rot=self.rot)
+            msk = self.get_mask()
+            mp[msk == 0] = 0
+            return mp
+
+        fn = '_'.join([f'{self.map_name}_{mod}_w2s2',
+                       f'coord{self.coords}',
+                       f'ns{self.nside}.fits.gz'])
+        w2s2 = self._rerun_read_cycle(fn, 'FITSMap', get_w2s2)
+
+        N_ell = hp.nside2pixarea(self.nside) * np.sum(w2s2) / self.npix
+        nl = N_ell * np.ones(3*self.nside)
+        nl[:2] = 0  # Ylm = for l < spin
+        self.nls[mod] = np.array([nl, 0*nl, 0*nl, nl])
+        self.nl_coupled = self.nls[mod]
+        return self.nl_coupled
+
+    def get_dtype(self):
+        return 'galaxy_shear'
+
+    def get_spin(self):
+        return 2
+
+
+# This function was used to shorten the official catalog. It is not used to
+# produce the Cells. It was tested apart and writing a test for it complicates
+# the test data generartion a bit. I'm going to skip it from coverage for now.
+def save_index_short_per_bin(path):  # pragma: no cover
```

Comment:
Just to clarify, this is here only in case we need something similar in the future, but it's not really used in the code?

Reply:
That is correct. Are we happy to exclude it from the tests, then?

Reply:
yes, that's fine